buildscripts/builddrivers.py

```diff
@@ -0,0 +1,287 @@
+        util            # BuildUtil object whose constructor takes phpver, driver, arch, thread, debug 
+        repo            # GitHub repository
+        branch          # GitHub repository branch
+        download_source # download source from GitHub or not
+        remote_path     # remote destination to where the drivers will be placed (None for local builds)
+        local           # a boolean flag - whether the build is local
+        rebuild         # a boolean flag - whether the user is rebuilding
+        make_clean      # a boolean flag - whether make clean is necessary
+        source_path     # path to a local source folder
+    """
+    
+    def __init__(self, phpver, driver, arch, thread, debug, repo, branch, download, path):
+        self.util = BuildUtil(phpver, driver, arch, thread, debug)
+        self.repo = repo
+        self.branch = branch
+        self.download_source = download
+        self.remote_path = path
+        self.local = path is None   # the default path is None, which means running locally 
+        self.rebuild = False
+        self.make_clean = False
+        self.source_path = None     # None initially but will be set later if not downloading from GitHub
+    
+    def show_config(self):
+        print()
+        print('PHP Version: ', self.util.phpver)
+        print('Arch: ', self.util.arch)
+        print('Thread: ', self.util.thread)
+        print('Driver: ', self.util.driver) 
+        print('Debug enabled: ', self.util.debug_enabled) 
+        print()
+
+    def clean_or_remove(self, root_dir, work_dir):
+        """Only check this when building locally and not rebuilding. If the php source directory 
+        already exists, this will prompt user whether to rebuild, clean, or superclean, the last option
+        will remove the entire php source directory.
+        
+        :param  root_dir: the C:\ drive
+        :param  work_dir: the directory of this script
+        :outcome: the old binaries, if exist, will be removed
+        """
+        phpsrc = self.util.phpsrc_root(root_dir)
+        if os.path.exists( phpsrc ):
+            print(phpsrc + " exists.")
+            print("Choose rebuild(r) if using the same configuration. Choose clean(c) otherwise. If unsure, choose superclean(s).") 
+            build_choice = validate_input("Want to rebuild (r), clean (c) or superclean (s)? ", "r/c/s")
+            self.make_clean = False
+            if build_choice == 'r':
+                print('Will rebuild the binaries')
+                # only the old binaries based on the current configuration will be removed
+                self.util.remove_prev_build(root_dir)
+            elif build_choice == 'c':
+                print('Will make clean')
+                self.make_clean = True
+                # all old builds are removed, and this step is necessary because 
+                # the user might have changed the configuration
+                self.util.remove_old_builds(root_dir)
```

Comment:
Does this option ever call 'nmake clean'? It seems that remove_old_builds removes the binaries but not the object files. Does PHP rebuild the object files if the configuration changes?